tests/unit/core/test_utils.py

```diff
@@ -330,3 +337,67 @@ def test_load_dot_env_loads_git_root_env(
     with cd(str(sub1_sub2_dir)):
         load_dot_env()
         load_dotenv_mock.assert_called_once_with(str(git_root_dotenv), override=True)
+
+
+@patch("requests.get")
+@pytest.mark.parametrize(
+    "current_version,remote_version,print_warning",
+    (
+        ("1.2.3", "1.2.4", True),
+        ("1.2.3", "1.3.3", True),
+        ("1.2.3", "2.2.3", True),
+        ("1.2.3", "1.2.3", False),
+        ("1.2.3", "1.2.2", False),
+        ("1.2.3", "1.1.3", False),
+        ("1.2.3", "0.2.3", False),
+    ),
+)
+def test_check_for_update(
+    request_get_mock: Mock,
+    current_version: str,
+    remote_version: str,
+    print_warning: bool,
+    capsys: CaptureFixture[str],
+    monkeypatch: MonkeyPatch,
+):
```

Comment:
You can use the `fs` fixture from [pyfakefs](https://pypi.org/project/pyfakefs/) in your test to ensure the test never touches the real file system. This avoids side-effects and removes the need for the optional `cache` argument in `check_for_update()`.